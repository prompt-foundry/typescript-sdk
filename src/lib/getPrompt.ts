import { Prompt, PromptMessageRole } from '../types/Prompt'

export const getPrompt = ({ promptId }: { promptId: string }): Promise<Prompt> => {
  return Promise.resolve({
    promptId,
    promptMessages: [
      {
        content: 'You are a helpful assistant.',
        role: PromptMessageRole.SYSTEM
      }
    ],
    promptParameters: {
      modelName: 'gpt-4o',
      temperature: 1,
      topP: 1,
      frequencyPenalty: 0,
      presencePenalty: 0,
      responseFormat: {
        type: 'text'
      }
    }
  })
}
