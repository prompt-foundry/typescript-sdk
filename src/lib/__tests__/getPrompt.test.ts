import { getPrompt } from '../getPrompt'

describe('getPrompt', () => {
  test('hello world test', () => {
    expect(getPrompt({ promptId: 'HELLO' })).toMatchSnapshot()
  })
})
