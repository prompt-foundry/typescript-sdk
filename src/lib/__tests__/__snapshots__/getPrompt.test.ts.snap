// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPrompt hello world test 1`] = `Promise {}`;
